src/data.rs

```diff
@@ -297,16 +366,50 @@ impl Builder for StringDataBuilder {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
+pub struct BinaryDataBuilder {}
+
+impl TryFrom<TypeBuilderContainer> for BinaryDataBuilder {
+    type Error = CryptoError;
+
+    fn try_from(builder: TypeBuilderContainer) -> Result<Self, Self::Error> {
+        match builder.0 {
+            TypeBuilder::Data(DataBuilder::Binary(bdb)) => Ok(bdb),
+            _ => Err(CryptoError::NotDowncastable),
+        }
+    }
+}
+
+impl From<BinaryDataBuilder> for TypeBuilder {
+    fn from(bdb: BinaryDataBuilder) -> TypeBuilder {
+        TypeBuilder::Data(DataBuilder::Binary(bdb))
+    }
+}
+
+impl Builder for BinaryDataBuilder {
+    type Output = Data;
+
+    fn build(&self, data: Option<&[u8]>) -> Result<Self::Output, CryptoError> {
+        match data {
+            Some(binary_data_bytes) => {
+                let s = String::from_utf8(binary_data_bytes.to_vec())
```

Comment:
This goes along with the review comment re using a String as the data type (unless there's something I'm missing here). Should just be able to keep `binary_data_bytes.to_vec()` and remove the wrapping `from_utf8`